src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalController.java

```diff
@@ -425,4 +431,84 @@ public void deletePaymentAccountsOfOrganisation(
 
     }
 
+    @ApiOperation(
+            value = "Adds contact informations(address details) to organisation",
+            notes = "**IDAM Roles to access API** :\n pui-organisation-manager",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            }
+
+    )
+    @ApiResponses({
+            @ApiResponse(
+                    code = 201,
+                    message = ""
+            ),
+            @ApiResponse(
+                    code = 400,
+                    message = "An invalid request has been provided"
+            ),
+            @ApiResponse(
+                    code = 401,
+                    message = "Unauthorized Error : The requested resource is restricted and requires authentication"
+            ),
+
+            @ApiResponse(
+                    code = 403,
+                    message = "Forbidden Error: Access denied"
+            ),
+            @ApiResponse(
+                    code = 500,
+                    message = "Internal Server Error"
+            )
+    })
+
+    @PostMapping(
+            path = "/addresses",
+            consumes = APPLICATION_JSON_VALUE,
+            produces = APPLICATION_JSON_VALUE
+    )
+    @ResponseStatus(value = HttpStatus.CREATED)
+    @ResponseBody
+    @Secured({"pui-organisation-manager"})
+    public ResponseEntity<Void> addContactInformationsToOrganisation(
+            @NotNull @RequestBody List<@Valid ContactInformationCreationRequest> contactInformationCreationRequests,
```

Comment:
will this cater  empty request or empty address list in request ?

Reply:
resolved